app/forms/supplementary/ProcedureCode.scala

```diff
@@ -41,32 +77,14 @@ case class ProcedureCodes(
 
     procedureCodeMapping ++ additionalProcedureCodesMapping
   }
-}
 
-object ProcedureCodes {
-  implicit val format = Json.format[ProcedureCodes]
+  def toProcedureCode(): ProcedureCode = ProcedureCode(procedureCode, None)
 
-  private val procedureCodeLength = 4
-  private val additionalProcedureCodeLength = 3
-  val mapping = Forms.mapping(
-    "procedureCode" -> text()
-      .verifying("supplementary.procedureCodes.procedureCode.error.empty", _.trim.nonEmpty)
-      .verifying(
-        "supplementary.procedureCodes.procedureCode.error.length",
-        isEmpty or hasSpecificLength(procedureCodeLength)
-      )
-      .verifying("supplementary.procedureCodes.procedureCode.error.specialCharacters", isAlphanumeric),
-    "additionalProcedureCodes" -> seq(
-      text()
-        .verifying(
-          "supplementary.procedureCodes.additionalProcedureCode.error.length",
-          isEmpty or hasSpecificLength(additionalProcedureCodeLength)
-        )
-        .verifying("supplementary.procedureCodes.additionalProcedureCode.error.specialCharacters", isAlphanumeric)
-    ).verifying("supplementary.procedureCodes.additionalProcedureCode.error.singleEmpty", _.exists(_.trim.nonEmpty))
-  )(ProcedureCodes.apply)(ProcedureCodes.unapply)
+  def contains(code: String): Boolean = additionalProcedureCodes.contains(code)
```

Comment:
This method's name is also not fully explanatory. At first I thought it would also look after Procedure Code.